### PR TITLE
[New Dashboard] Changes related to pipeline operation related buttons

### PR DESCRIPTION
**Pull request by yankurk**
#4348

### CODE REVIEW

**Comment by yankurk** [comment]
As part of this PR:
* Trigger buttons are disabled if a pipeline is paused
* Trigger and pause/unpause buttons are not shown if user doesn't have operating rights

File: server/webapp/WEB-INF/rails.new/spec/webpack/views/dashboard/pipeline_widget_spec.js [hunk 1]
```diff
@@ -540,6 +547,17 @@ describe("Dashboard Pipeline Widget", () => {
         const modalTitle = $('.modal-title:visible');
         expect(modalTitle).toHaveText(`${pipeline.name} - Trigger`);
       });
+
+      it("should disable button when pipeline is paused", () => {
+        unmount();
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/spec/webpack/views/dashboard/pipeline_widget_spec.js]: Can we not mount in a setup method if it requires us to unmount before a test?

File: server/webapp/WEB-INF/rails.new/webpack/views/dashboard/pipeline_operations_widget.js.msx [hunk 2]
```diff
@@ -148,26 +148,33 @@ const PipelineOperationsWidget = {
     }
 
     let isTriggerDisabled = false;
-    if (!pipeline.canOperate || pipeline.isFirstStageInProgress() || pipeline.isLocked) {
+    if (!pipeline.canOperate || pipeline.isFirstStageInProgress() || pipeline.isLocked || pipeline.isPaused) {
       isTriggerDisabled = true;
     }
 
+    let operationRelatedButtons = "";
+    if(pipeline.canAdminister || pipeline.canOperate) {
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/webpack/views/dashboard/pipeline_operations_widget.js.msx]: Run lint?

**Comment by yankurk** [comment]
There is one case I overlooked in this PR. It is related to this [commit ](https://github.com/gocd/gocd/commit/5caef8ecee0c71c5bbed3d2a6e46930fd4f0201f) which says that if there are operators defined for the first stage of a pipeline, those operators override the operators defined in the pipeline group level. However, the operators defined at pipeline group level can still pause the pipeline in such scenarios. This means that trigger and pause are treated differently. I'm working on a fix for such scenarios on the new dashboard.

**Comment by yankurk** [comment]
These are the states that operation buttons can be in the new dashboard:
1) For users with operate permissions (and by default for admins):
screen shot 2018-02-20 at 12 11 08 pm

2) For users with pause permission but no operate permission (see 5caef8ecee0c71c5bbed3d2a6e46930fd4f0201f)
screen shot 2018-02-20 at 12 12 37 pm

3) For users with only view permission
screen shot 2018-02-20 at 12 14 58 pm

4) Trigger buttons disabled due to pipeline being paused
screen shot 2018-02-20 at 12 11 35 pm

5) Trigger buttons disabled because the first stage is still in progress
screen shot 2018-02-20 at 12 12 00 pm

**Comment by ketan** [comment]
@yankurk — Removing buttons from the page does not feel right, because users don't know why they do not see the button. It's better to show the button, but have it disabled, preferably with a tooltip to say why it's been disabled.

In most cases, when something does not show up, users may think that they have done something wrong, or there's a bug with the system.

**Comment by yankurk** [comment]
@ketan -- The decision to remove the buttons is also tied to whether we show pipeline and pipeline group edit gear icon for non admin users #4375. If we decide to keep the trigger buttons in disabled state, we need to keep the edit gear icon in disabled state for non-admins too.

**Comment by GaneshSPatil** [comment]
@yankurk -- I guess, we decided not to remove any element and disable them based on the permissions.
Same applies to settings gear and operation buttons too.

**Comment by maheshp** [comment]
I was thinking bout guest users on build.gocd, there is no way a guest user will be able to operate or edit a pipeline. In these cases wondering if it makes sense to hide those buttons over disabling them?

**Comment by arvindsv** [comment]
As long as it is clear why it is disabled, in the tool tip, I think it's fine to keep them. Showing a proper reason ("You do not have permission" vs "This pipeline is about to trigger") might not be very easy.

**Comment by naveenbhaskar** [comment]
Disabling and hiding elements 
case 1:  User has no way to do anything with the elements. In that case, we dont have to show the element. If we show the element and then disable it, that means, user can do something to make it enabled. for eg: config button can be used only with permission. if there is no permission for a user, then lets not show it.

case 2 : user can do some action to make the button enabled. In this case, button has to be visible

if there is something which cant be used by the user, there is no point in showing that. even if we show a tooltip there, its just an element in the UI which is not usable at all.

**Comment by arvindsv** [comment]
@naveenbhaskar The action that a user can perform to make it enabled is to convince an admin they need it. :)

GoCD's old dashboard, for a view user, looks quite bare and it doesn't show the possibilities that exist. I'd strongly consider disabling it, rather than hiding it. If we're hiding it, we can hide it by changing `opacity` to 0, so that the box sizes won't get affected, etc.

Finally, it is your call though, since you need to consider consistency (gear icon, etc).

**Comment by GaneshSPatil** [comment]
Closing this PR as we have decided to show the buttons and disabled message tooltip on hover

**Pull request closed without merging**